Frameworks/CoreFoundation/Preferences.subproj/CFPreferences.c

```diff
@@ -882,11 +882,7 @@ static Boolean synchronizeApplicationDomain(CFTypeRef context, void *domain) {
     CFDataRef data = CFPropertyListCreateData(__CFPreferencesAllocator(), (CFPropertyListRef)domain, kCFPropertyListXMLFormat_v1_0, 0, NULL);
     Boolean success = false;
     if (data) {
-#if DEPLOYMENT_TARGET_WINDOWS
-        success = _CFWriteBytesToFileAsync((CFURLRef)context, CFDataGetBytePtr(data), CFDataGetLength(data));
```

Comment:
I could not repro the reported issue when exiting out of the app normally (through the x button), and could only repro through debugger-stopping the app. This would not help the latter case. Additionally, judging by https://medium.com/@hanru.yeh/nsuserdefaults-is-planned-to-deprecated-cc185e19e6f8, it seems that [NSUserDefaults synchronize] is blocking on the reference platform.